pkg/runctl/cfg/pipelinerun_accessor.go

```diff
@@ -0,0 +1,59 @@
+package cfg
+
+import (
+	"github.com/SAP/stewardci-core/pkg/apis/steward/v1alpha1"
+	yaml "gopkg.in/yaml.v3"
+)
+
+// PipelineRunAccessorConfig is the config to defines a way to access a value of a pipeline run
+type PipelineRunAccessorConfig struct {
+	Kind Kind   `yaml:"kind,omitempty"`
+	Name string `yaml:"name,omitempty"`
+}
+
+// PipelineRunAccessor defines an interface to access a pipeline run
+type PipelineRunAccessor interface {
+	Access(run *v1alpha1.PipelineRun) string
+}
```

Comment:
I would go for a simple function type:

```go
type extractor func(run *v1alpha1.PipelineRun) any
```

Note that the return value is of type `any`, because log parameters can have any type.

Reply:
See https://github.com/SAP/stewardci-core/pull/412/commits/14c38499a2a7353e2d62333bb349aa56aba6dccb